src/application_settings/container_base.py

```diff
@@ -112,74 +104,149 @@ def filepath(cls) -> PathOpt:
         """Return the path for the file that holds the config / settings."""
         return _ALL_PATHS.get(id(cls), cls.default_filepath())
 
-    @classmethod
-    def _get(cls: type[_ContainerT]) -> Optional[_ContainerT]:
-        """Private getter for the singleton."""
-        return _ALL_CONTAINERS.get(id(cls))
+    if sys.version_info >= (3, 11):
```

Comment:
I'm not really sure why this is now back? So we again have multiple implementations, but I don't see any difference?

Reply:
so yeah a couple of comments placed but I skipped most of it, again I think we should move all this version dependent stuff out of this file itself, and create type aliases at the top of this file, or in another file.

Reply:
In 3.11 Self has been introduced; for versions below that, we need to define a TypeVar for that. A TypeVar is a _variable_ that needs to be binded, either to cls (via cls: type[Self]) or to self (via self: Self) so that the TypeVar actually represents the corresponding type. In 3.11, this binding is automatic and you get an error if you do specify something. You cannot replace this situation with a type alias, unfortunately. Hence, for this Self typing, afaik we are stuck with multiple implementations here...

Reply:
@tjadevries I sort of get that, but I still don't get why we need the if statements? As far as I can see they are exactly identical? (only `Self` is either imported, or is defined as `TypeVar("Self", "ContainerBase")` (or another class).

The only difference I'm seeing is an untyped `cls` vs a `Type[Self]` cls. Can't we just remove the `Type[Self]`?

Reply:
Okay it indeed seems not entirely trivial. Not entirely sure if it's possible but I think we should now just limit ourselves to the lowest version that we support. I expect that `python3.11` should still support all the typing that is possible in `python3.9` right? So in my opinion we should just not use `Self` until we drop support for python 3.10.

These double methods just because of the typing is not really acceptable I think.